### ISSUE TITLE
excuse me, can you provide the best model of vcoco? Thanks!

### ISSUE THREAD

**Issue by whyang78** [fredzzhang/spatially-conditioned-graphs#70]

**Comment by fredzzhang** [comment]
Hi @whyang78,

We didn't attach the V-COCO model because it takes very little time to train it. Based on the instructions, you should be able to reproduce the results fairly easily. Were you able to do so?

Fred.

**Comment by whyang78** [comment]
i set bs=4 and 4 gpus , learning rate=0.00005,i find the map of vcoco is 50.3. can you solve this problem， thank you

**Comment by whyang78** [comment]
hi，thank you for your answers！because my computer is not able to use batchsize=32，so it is difficult to reproduce the reaults.or can you provide the vcoco cache results of best model? thank you very much !

**Comment by fredzzhang** [comment]
Hi @whyang78,

The batch size is not a big problem. Your setup should be fine. One thing to point out is that, a few of the V-COCO action classes do not correspond to HOIs. Based on the number you posted, it is likely that you did not exclude those classes. 

![Screen Shot 2021-11-12 at 10 42 30 am](https://user-images.githubusercontent.com/11484831/141384499-7f9c987c-9a0b-49c9-8893-926c0944b153.png)

The image above is the results from [InteractNet](https://openaccess.thecvf.com/content_cvpr_2018/papers/Gkioxari_Detecting_and_Recognizing_CVPR_2018_paper.pdf). Classes included in the red box are the ones you should average over. Make sure you do the same.

If I remember this correctly, the V-COCO script computes AP for a redundant class _point_. The AP for that class should be zero since it is never predicted. The number you get is the mAP for 25 classes with one of them being 0. So the correct performance should be 50.3 * 25 / 24 = 52.4. The more rigorous way to compute this is to sum up the APs for the 24 classes and compute the mean.

Let me know how it goes.

Fred.

**Comment by whyang78** [comment]
> Hi @whyang78,
> 
> The batch size is not a big problem. Your setup should be fine. One thing to point out is that, a few of the V-COCO action classes do not correspond to HOIs. Based on the number you posted, it is likely that you did not exclude those classes.
> 
> ![Screen Shot 2021-11-12 at 10 42 30 am](https://user-images.githubusercontent.com/11484831/141384499-7f9c987c-9a0b-49c9-8893-926c0944b153.png)
> 
> The image above is the results from [InteractNet](https://openaccess.thecvf.com/content_cvpr_2018/papers/Gkioxari_Detecting_and_Recognizing_CVPR_2018_paper.pdf). Classes included in the red box are the ones you should average over. Make sure you do the same.
> 
> If I remember this correctly, the V-COCO script computes AP for a redundant class _point_. The AP for that class should be zero since it is never predicted. The number you get is the mAP for 25 classes with one of them being 0. So the correct performance should be 50.3 * 25 / 24 = 52.4. The more rigorous way to compute this is to sum up the APs for the 24 classes and compute the mean.
> 
> Let me know how it goes.
> 
> Fred.

thank you very much! I will try again!
